File: MusicWeb.Server/MusicWeb.Models/Entities/Users/UserFavoriteSong.cs
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Identity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities
{
    public class UserFavoriteSong : BaseEntity
    {
        public DateTime FavoriteDate { get; set; }

        public string UserId { get; set; }
        public int SongId { get; set; }
        public virtual Song Song { get; set; }
        public virtual ApplicationUser User { get; set; }
    }
}

File: MusicWeb.Client/src/models/UserFavoriteSong.js
export default class UserFavoriteSong {
  id = null;
  userId = null;
  songId = null;
  song = {};
  user = {};
  songId = null;
  constructor($data) {
    if ($data) {
      this.id = $data.id;
      this.userId = $data.userId;
      this.songId = $data.songId;
      this.song = $data.song ? $data.song : {};
      this.user = $data.user ? $data.user : {};
    }
  }
}
